src/config.h

```diff
@@ -19,99 +19,927 @@
  */
 
 #pragma once
-#ifndef CONFIG_H
-#define CONFIG_H
+#ifndef CONFIG_MAIN_H
+#define CONFIG_MAIN_H
 
-#include <string>
+#include <PGE_File_Formats/file_formats.h>
 
-extern struct Config_t
+#include "sdl_proxy/sdl_audio.h"
+#include "sound.h"
+#include "screen.h"
+
+#include "Logger/logger_level.h"
+
+#include "config/config_base.hpp"
+#include "config/config_hooks.h"
+
+struct saveUserData;
+
+template<bool writable>
+class _Config_t
 {
+public:
+    struct Scope
+    {
+        enum
+        {
+            None = 0,
+            UserGlobal = 1 << 0,
+            UserEpisode = 1 << 1,
+            UserFile = 1 << 2,
+
+            UserLocal = UserEpisode | UserFile,
+            User = UserGlobal | UserLocal,
+
+            CreatorGlobal = 1 << 3,
+            CreatorEpisode = 1 << 4,
+            CreatorFile = 1 << 5,
+
+            CreatorLocal = CreatorEpisode | CreatorFile,
+            Creator = CreatorGlobal | CreatorLocal,
+
+            Episode = UserEpisode | CreatorEpisode,
+            File = UserFile | CreatorFile,
+            Local = Episode | File,
+            Global = UserGlobal | CreatorGlobal,
+            All = 0xff,
+        };
+    };
```

Comment:
Maybe use the `enum class` here? C++11 introduces this feature here.

Reply:
`enum class` doesn't support bitwise operations. But, I think I can simply this whole system now, with just three values (user-settable, assets-settable, episode-settable).